### DISCUSSION TITLE
Predictable intersections discovery

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#665] [Q&A] [answered]
The following code produces the figure below. I was able to find out the labels for the intersections anchors only by _trial and error_. Is there some way of getting the labels of those intersections by using a systematic and deterministic way (and not by _trial and error_)? Thanks!

```
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *
 
  intersections("i", {
    circle((-3,0), radius: 0.5)
    circle((rel: (6,0)), radius: 0.5)
    circle((rel: (-3,3)), radius: 0.5)
    circle((rel: (0,-6)), radius: 0.5)
    hide(circle((0,0), radius: 3))
  })

  arc-through("i.0", (135deg, 3) , "i.5")
  arc-through("i.4", (45deg, 3) , "i.3") 
  arc-through("i.6", (-45deg, 3) , "i.2")
  arc-through("i.1", (-135deg, 3) , "i.7")

  for-each-anchor("i", (name) => {
    circle("i." + name, radius: 0.1, fill: red, stroke: red)
  })
})
```

![image](https://github.com/user-attachments/assets/7de7b941-dde9-4cb7-9cf8-63a2861251fe)

**Comment by fenjalien** [comment, marked as answer]
At the moment there is no sorting so it is difficult to tell. However, if you use `for-each-anchor` and display the anchor's name at its position it should tell you easily enough.
```typ
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *
 
  intersections("i", {
    circle((-3,0), radius: 0.5)
    circle((rel: (6,0)), radius: 0.5)
    circle((rel: (-3,3)), radius: 0.5)
    circle((rel: (0,-6)), radius: 0.5)
    hide(circle((0,0), radius: 3))
  })

  arc-through("i.0", (135deg, 3) , "i.5")
  arc-through("i.4", (45deg, 3) , "i.3") 
  arc-through("i.6", (-45deg, 3) , "i.2")
  arc-through("i.1", (-135deg, 3) , "i.7")

  for-each-anchor("i", (name) => {
    content("i." + name, name, fill: white, frame: "rect", stroke: white)
  })
})
```

![image](https://github.com/user-attachments/assets/2943493d-d5d7-4381-ba94-68cd575b44c4)

**Reply by psads-git** [reply]
Thanks, @fenjalien, for your reply! While the workaround you suggest is a nice solution when dealing with a modest number of circles, it could become cumbersome if the number of circles is very large. It might be advisable to consider introducing some form of anchor sorting.

**Reply by fenjalien** [reply]
Yes we're thinking about it, you could make an issue about it so we could keep track.

**Reply by psads-git** [reply]
Done so, @fenjalien: https://github.com/cetz-package/cetz/issues/667
